packages/manager/apps/pci-savings-plan/src/components/Dashboard/Kpis/Kpis.tsx

```diff
@@ -61,15 +61,24 @@ const Kpis = ({
   consumption: SavingsPlanFlavorConsumption;
 }>) => {
   const { t } = useTranslation('dashboard');
+  const defaultMessage = t('dashboard_kpis_not_available');
 
-  const computedUsagePercent = useMemo(
-    () =>
-      formatUsagePercent(
-        getUsagePercent(consumption),
-        t('dashboard_kpis_not_available'),
-      ),
-    [consumption, t],
-  );
+  const computedPercents = useMemo(() => {
+    if (!consumption?.periods?.length) {
```

Comment:
```suggestion
    if (!consumption.periods?.length) {
```
Consumption cannot be null or undefined, does periods can also be null or undefined ?

Reply:
yes, it could